src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/WorkflowController.java

```diff
@@ -72,18 +76,44 @@
     @ResponseStatus(HttpStatus.CREATED)
-    public WorkflowMetadata create(@PathVariable Long bucketId,
+    public WorkflowMetadataList create(@PathVariable Long bucketId,
             @ApiParam(value = "Layout describing the tasks position in the Workflow") @RequestParam(required = false) Optional<String> layout,
+            @ApiParam(value = "Import workflows from ZIP when set to 'zip'.") @RequestParam(required = false) Optional<String> alt,
             @RequestPart(value = "file") MultipartFile file) throws IOException {
-        return workflowService.createWorkflow(bucketId, layout, file.getBytes());
+        if (alt.isPresent() && "zip".equals(alt.get())) {
+            return new WorkflowMetadataList(workflowService.createWorkflows(bucketId, layout, file.getBytes()));
+        } else {
+            return new WorkflowMetadataList(workflowService.createWorkflow(bucketId, layout, file.getBytes()));
+        }
     }
 
     @ApiOperation(value = "Gets a workflow's metadata by IDs", notes = "Returns metadata associated to the latest revision of the workflow.")
     @ApiResponses(value = @ApiResponse(code = 404, message = "Bucket or workflow not found"))
-    @RequestMapping(value = "/buckets/{bucketId}/workflows/{workflowId}", method = GET)
-    public ResponseEntity<?> get(@PathVariable Long bucketId, @PathVariable Long workflowId,
-            @ApiParam(value = "Force response to return workflow XML content when set to 'xml'.") @RequestParam(required = false) Optional<String> alt) {
-        return workflowService.getWorkflowMetadata(bucketId, workflowId, alt);
+    @RequestMapping(value = "/buckets/{bucketId}/workflows/{idList}", method = GET)
+    public ResponseEntity<?> get(@PathVariable Long bucketId, @PathVariable List<Long> idList,
+            @ApiParam(value = "Force response to return workflow XML content when set to 'xml'. Or extract workflows as ZIP when set to 'zip'.") @RequestParam(required = false) Optional<String> alt,
+            HttpServletResponse response) throws MalformedURLException {
+        if (alt.isPresent() && "zip".equals(alt.get())) {
+            byte[] zip = workflowService.getWorkflowsAsArchive(bucketId, idList);
+
+            response.setStatus(HttpServletResponse.SC_OK);
+            response.setContentType("application/zip");
+            response.addHeader("Content-Disposition", "attachment; filename=\"archive.zip\"");
+            response.addHeader("Content-Transfer-Encoding", "binary");
+            try {
+                response.getOutputStream().write(zip);
+                response.getOutputStream().flush();
+            } catch (IOException ioe) {
+                throw new RuntimeException(ioe);
+            }
+            return ResponseEntity.ok().build();
+        } else {
+            if (idList.size() >= 1) {
+                return workflowService.getWorkflowMetadata(bucketId, idList.get(0), alt);
+            } else {
+                throw new MalformedURLException();
```

Comment:
I would have returned a response with an appropriate HTTP status code (405?).

Below is an interesting diagram to make the decision:

https://github.com/for-GET/http-decision-diagram